src/utils/DoubleAggregator.hpp

```diff
@@ -0,0 +1,72 @@
+#include <boost/accumulators/accumulators.hpp>
+#include <boost/accumulators/statistics.hpp>
+#include <boost/accumulators/statistics/sum_kahan.hpp>
+
+namespace precice::utils {
+
+/// An accurate aggregator for doubles with usability in mind
+class DoubleAggregator {
+private:
+  using Acc = boost::accumulators::accumulator_set<double, boost::accumulators::stats<boost::accumulators::tag::sum_kahan>>;
```

Comment:
```suggestion
  /// This class is mainly a wrapper for boost's Kahan accumulator which implements the Kahan summation algorithm (see https://en.wikipedia.org/wiki/Kahan_summation_algorithm)
  using Acc = boost::accumulators::accumulator_set<double, boost::accumulators::stats<boost::accumulators::tag::sum_kahan>>;
```
I think it would be nice to also mention the keyword "wrapper" here and provide a brief link to what we actually do. Both are implementation details. Therefore, I did not put it into the class description above. Not sure that this is 100% the right approach. Maybe you have a better idea @fsimonis?

Reply:
Good idea.